settings-preferences: Create left sidebar section.

shuklamaneesh23 opened the pull request:
This commit removes the following settings from the Settings/Preferences Information section and adds a new section in the Settings/Preferences called as Left Sidebar section:

- Show unread counts for
- Show unread count total on home view
- Show counts for starred messages
- Group channels by folder in the left sidebar
- Hide inactive channels

Fixes #35601.

<!-- Describe your pull request here.-->


<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

| Before | After |
|---|--|
| <img width="563" height="756" alt="Screenshot 2025-08-05 at 2 14 36 PM" src="https://github.com/user-attachments/assets/a8fc0877-3e49-4221-806d-e8efd5420afe" /> | <img width="546" height="510" alt="Screenshot 2025-08-05 at 2 03 50 PM" src="https://github.com/user-attachments/assets/e2bf74b5-b1ad-4365-acea-ef4925f0c2c9" /> |
| <img width="563" height="756" alt="Screenshot 2025-08-05 at 2 14 36 PM" src="https://github.com/user-attachments/assets/a8fc0877-3e49-4221-806d-e8efd5420afe" /> | <img width="532" height="340" alt="Screenshot 2025-08-05 at 2 03 55 PM" src="https://github.com/user-attachments/assets/c0a34760-a33d-4864-9d25-79b65d6e9702" /> |


<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

shuklamaneesh23 commented:
@zulipbot add "buddy review"

shuklamaneesh23 commented:
Hey @alya , I've updated the help-docs as per the change. 
Please have another look.
Thanks!

alya commented on help/manage-inactive-channels.md:
```diff
@@ -48,7 +48,7 @@ channels if you need to access them.
 {settings_tab|preferences}
 
-1. Under **Information**, configure **Hide inactive channels**.
+1. Under **Left Sidebar**, configure **Hide inactive channels**.
```
```suggestion
1. Under **Left sidebar**, configure **Hide inactive channels**.
```

shuklamaneesh23 commented:
There are 5 settings which I moved from the Information section to a separate left sidebar section and there are a total of 5 places where I changed the Information with the Left sidebar in the help docs.

timabbott commented:
Merged, thanks @shuklamaneesh23 and @alya! @sahil839 FYI :)
